packages/sn-filter-pane/src/components/FoldedListbox/SelectionSegmentsIndicator.tsx

```diff
@@ -1,47 +1,37 @@
 import React from 'react';
 import { Grid } from '@mui/material';
 import getSegmentsSizes from './get-segment-sizes';
+import { ISelectionsComponent, IStyles } from '../../hooks/types/components';
 
 interface SelectionSegmentsIndicatorProps {
   qDimensionInfo: EngineAPI.INxDimensionInfo,
   fullHeight: boolean,
+  styles?: IStyles,
 }
 
-const SelectionSegmentsIndicator = ({ qDimensionInfo, fullHeight }: SelectionSegmentsIndicatorProps) => {
+type SelectionState = ('selected' | 'selectedExcluded' | 'possible' | 'alternative' | 'excluded');
+
+const SelectionSegmentsIndicator = ({ qDimensionInfo, fullHeight, styles }: SelectionSegmentsIndicatorProps) => {
   if (!qDimensionInfo) {
     return null;
   }
   const ratios = getSegmentsSizes(qDimensionInfo);
   const segmentHeight = fullHeight ? '100%' : '4px';
-  const segments = [
-    {
-      key: 'green',
-      style: {
-        background: '#009845',
-        height: segmentHeight,
-        width: ratios.green,
-      },
-    },
-    {
-      key: 'white',
-      style: {
-        background: '#fff',
-        height: segmentHeight,
-        width: ratios.white,
-      },
-    },
-    {
-      key: 'grey',
-      style: {
-        background: '#d2d2d2',
-        height: segmentHeight,
-        width: ratios.grey,
-      },
+  const { selections = {} as ISelectionsComponent['colors'] } = styles?.listbox || {};
+
+  // The keys also determine the order in which the sections will be rendered.
+  const keys: SelectionState[] = ['selected', 'selectedExcluded', 'possible', 'alternative', 'excluded'];
```

Comment:
If `as const` is added to this array, the type `SelectionState` can be removed.
```suggestion
  const keys = ['selected', 'selectedExcluded', 'possible', 'alternative', 'excluded'] as const;
```